### ISSUE TITLE
Provide function that checks if hook environment is already installed.

### ISSUE THREAD

**Issue by krzyslom** [lorenzwalthert/precommit#159]
This is semi related to #158 . The `use_precommit` documentation says:

>When to call this function?
...
You cloned a repo that has a .pre-commit-config.yaml already. You need to make sure git calls the hooks before the next commit.

One way to automate this step is to put `precommit::use_precommit(open = FALSE)` call inside the `.First` function in the Rprofile. However this will result in calling `use_precommit` every time the user opens the project, which results in ~8 second delay.

I wonder if there is a way to check if the environment has been already installed. Then `use_precommit` function could be called only if e.g. `is_hooks_env_installed()` returns `TRUE`. Unfortunately at the moment I'm not able to pinpoint what changes after hooks env installation and what could be tested against with the function.

**Comment by lorenzwalthert** [comment]
I am not sure why we need to automate this in {precommit}. What's your usecase? If you clone repos at scale, you already have a script to do the cloning, so I don't see why not add a `precommit::use_precommit()` call, or alternatively just bare `$ pre-commit install` to omit the `autoupdate()`? The latter should probably only run when there is no config already in the repo, but le'ts discuss that in another issue. 

Also, I think such initialisation, even when conditional, should not go into the `.Rprofile` because it's really a one time thing you do at cloning, but that's probably personal taste . I don't know how to implement `is_hooks_env_installed()` anyways and I think this is functionality that belongs to upstream pre-commit if anywhere. I don't want to fiddle with the internals of pre-commit to determine if hooks are installed. I am happy to export an R wrapper if this API change is accepted there though.

**Comment by lorenzwalthert** [comment]
Also please see that you can git provides a mechanism to put some files in your `.git` folder at creation time, so I think you can skip the `$ pre-commit install` step with 
```bash
git config --global init.templateDir ~/.git-template
pre-commit init-templatedir ~/.git-template
```

according to https://pre-commit.com/#pre-commit-init-templatedir. Have not tested it properly and it probably also happens when you clone repos that don't use pre-commit and I don't know if that's really what you want, but maybe have a look.

**Comment by lorenzwalthert** [comment]
Note this approach with `init.templateDir` may give pre-commit troubles finding the executable.

**Comment by krzyslom** [comment]
>What's your usecase? If you clone repos at scale, you already have a script to do the cloning, so I don't see why not add a `precommit::use_precommit()` call, or alternatively just bare `$ pre-commit install` to omit the `autoupdate()`?

I'm creating a Docker image with particular set of tools. Multiple users will pull it and spin up their own containers. I can have precommit package installed in advance. With changes made in #158 I can also have hooks already installed in the conda env. Now every time a user manually clones a repository that already has yaml config file, he has to call once `precommit::use_precommit()` to install pre-commit inside the `.git/hooks` directory of the cloned repo. I want to take this responsibility from end user's shoulders.

Manipulations with git config could be troublesome since there are major host machine os related issues with files shared through Docker bind option. So if someone is mounting his host machine git config file inside the container it could get tricky.

>Also, I think such initialisation, even when conditional, should not go into the .Rprofile because it's really a one time thing you do at cloning, but that's probably personal taste .

I agree that such approach seems odd. Trying to run an operation multiple times when it's designed to be run only once seems impractical. I guess that this is a price to pay for trying to automate even this one single call. I've managed to come up with the following solution:
```R
are_hooks_installed <- function() {
  fs::file_exists(here::here(".git/hooks/pre-commit"))
}
```
On opening the cloned repo project, if this returns `FALSE`, then run `precommit::use_precommit()`, if returns `TRUE`, do nothing.

At the moment I'm satisfied with this approach. This function could be added to the package as a convenience function, but I totally understand if you recognize it as a unique corner case that would rarely be used and is not worth embedding in the package, specially when `precommit::autoupdate()` call could be conditional.

**Comment by lorenzwalthert** [comment]
Ok, thanks for explaining. I agree with the problems regarding git config, but did not mean to fiddle with the host machine. Can't this be done inside the docker container? I.e. somewhere in a `RUN` key in the `Dockerfile`? I thought the cloning and later committing happens within the container. 

In any case, I'd prefer if you'd implement this logic in our `.Rprofile`.

**Comment by krzyslom** [comment]
During the image build I can install precommit and all the other tools with RUN commands placed inside the Dockerfile. After that user downloads the image, spins up the container based on this image, opens web browser, and enters RStudio Server. Now inside the interactive R session he can clone repos that he wants to work with. I can't determine which repos will get cloned by the user in advance, but I want to use precommit on every single one that gets cloned and opened. Ideally this should be automated.

Thank you for your feedback. If think that this issue can be closed then.
